README.md

```diff
@@ -118,27 +58,48 @@ const pastaRevisionResponse = await getPageRevisionCountTimeseries(
 */
 ```
 
-## How to use the highlight functions
+## Highlighting
 
-### The Char Version vs Word Version
+Highlighting needs to be given raw content (text, including bolding and links). To highlight something, we need a context dictionary, with what is supposed to be highlighted, the content after and before. 
 
-To run the char version import the following in root.js or a file that is eventually imported in root.js to get the marks created in the html.
-```javascript
-import { text } from "./tagEveryChar";
-import { markPageChar, removeMarks } from "./markPageChar";
-``` 
+### Set the desired highlighting method
 
-To run the word version import the following in root.js or a file that is eventually imported in root.js to get the marks created in the html.
-```javascript
-import { text } from "./tagEveryWord";
-import { markPageWord, removeMarks } from "./markPageWord";
-``` 
+We have 3 highlighting methods (HighlightType.NODE, HighlightType.TAGGING_CHAR, HighlightType.TAGGING_WORD). Each one has its pros and cons. For more info, look at the enums.js file, and read the next sections.
+
+To change methods, go to ```root.js``` and change the var ```HIGHLIGHT_TYPE```
+
+### Highlighting - Node Version
+
+In this method, we go over the DOM tree text nodes searching for what should be highlighted, and only do so if the context before and after matches. There are several edge cases, so sometimes we use short context before/after. This type of highlighting is not ideal when the content to be highlighted is in different text nodes.
 
-Make sure that not both are being used at the same time and for best performance disable the old highlighting mechanism, which is easiest done by commenting out
 ```javascript
-import "./content";
+const arr = [
+  {
+    content_before: "800-acre ",
+    highlight: "Cowling Arboretum",
+    content_after: ", which",
+  },
+  {
+    content_before: "Teaching by ",
+    highlight: "U.S. News & World Report",
+    content_after: " for over a decade",
+  },
+  {
+    content_before: "was ",
+    highlight: "founded in",
+    content_after: " 1866",
+  },
+];
+
+let arrayLength = arr.length;
+for (let i = 0; i < arrayLength; i++) {
+  highlightContentUsingNodes(arr[i], "#99FF84");
+}
 ```
-in root.js. Note, this will disable the graph as well.
+
+## Highlighting - Char Version and Word Version
+
+This method will tag the entire wikipedia page, and we can highlight the given text by setting some CSS matching the given class. We can do highlighting in one pass, but it may be slower than the node version, however shouldn't be a problem in regular Wikipedia pages. For heavy pages, there's a way to not tag the references at the bottom of the page. 
```

Comment:
is a way not to tag references being implemented or mentioned somewhere? (remember @etiennerichart mentioning something about this)

Reply:
There's is code, but not in use. It's commented.